features/tag_sets/tag_set_associations.feature

```diff
@@ -54,3 +54,72 @@ Feature: Reviewing tag set associations
+      And a canonical character "Nathan Stark" in fandom "Eureka"
+      And a synonym "Nathan Carter" of the tag "Nathan Stark"
+      And I am logged in as "tagsetter"
+      And I set up the nominated tag set "Non-canonical Associations" with 1 fandom nom and 2 character noms
+      And I nominate fandom "Eureka" and characters "Jack Carter,Nathan Carter" in "Non-canonical Associations" as "tagsetter"
+    When I review nominations for "Non-canonical Associations"
+      And I check "fandom_approve_Eureka"
+      And I check "character_approve_Jack_Carter"
+      And I check "character_approve_Nathan_Carter"
+      And I press "Submit"
+      And I review associations for "Non-canonical Associations"
+    Then I should see "Nathan Carter → Eureka"
+
+  Scenario: When a nominated non-canonical is renamed, its associations remain for review
+    Given a canonical character "Jack Carter" in fandom "Eureka"
+      And a canonical character "Nathan Stark" in fandom "Eureka"
+      And a synonym "nathan carter" of the tag "Nathan Stark"
+      And I am logged in as "tagsetter"
+      And I set up the nominated tag set "Non-canonical Associations" with 1 fandom nom and 2 character noms
+      And I nominate fandom "Eureka" and characters "Jack Carter,nathan carter" in "Non-canonical Associations" as "tagsetter"
+    When I am logged in as a tag wrangler
+      And I edit the tag "nathan carter"
+      And I fill in "Name" with "Nathan Carter"
+      And I press "Save changes"
+    Then I should see "Tag was updated"
+    When I am logged in as "tagsetter"
+      And I review nominations for "Non-canonical Associations"
+      And I check "fandom_approve_Eureka"
+      And I check "character_approve_Jack_Carter"
+      And I check "character_approve_nathan_carter"
+      And I press "Submit"
+      And I review associations for "Non-canonical Associations"
+    Then I should see "Nathan Carter → Eureka"
+
+  Scenario: Nominating a new tag in a fandom generates associations for review
+    Given a canonical fandom "Eureka"
+      And I am logged in as "tagsetter"
+      And I set up the nominated tag set "New Associations" with 1 fandom nom and 2 character noms
+      And I nominate fandom "Eureka" and character "Jack Stark" in "New Associations" as "tagsetter"
+    When I review nominations for "New Associations"
+      And I check "fandom_approve_Eureka"
+      And I press "Submit"
+      And I edit nominations for "tagsetter" in "New Associations" to include characters "Jack Stark,Nathan Stark" under fandom "Eureka"
+      And I review nominations for "New Associations"
+    Then I should see "Jack Stark"
+      And I should see "Nathan Stark"
+    When I check "character_approve_Jack_Stark"
+      And I check "character_approve_Nathan_Stark"
+      And I press "Submit"
+      And I review associations for "New Associations"
+    Then I should see "Jack Stark → Eureka"
+      And I should see "Nathan Stark → Eureka"
```

Comment:
This bit specifically tests for the issue you found @sarken (I ran this with my old code and it failed properly). The smallest reproducible issue was
1. Approve the fandom but not everything in the nomination
2. Add a new tag that wasn't in the db before
3. Boom, no parent on the tag in part 2

Reply:
Thank you for figuring out the simplest form!